pkg/alertmanager/multitenant.go

```diff
@@ -97,12 +97,23 @@ type MultitenantAlertmanagerConfig struct {
 	FallbackConfigFile string `yaml:"fallback_config_file"`
 	AutoWebhookRoot    string `yaml:"auto_webhook_root"`
 
-	Store AlertStoreConfig `yaml:"storage"`
+	Store   AlertStoreConfig `yaml:"storage"`
+	Cluster ClusterConfig    `yaml:"cluster"`
 
 	EnableAPI bool `yaml:"enable_api"`
 }
 
-const defaultClusterAddr = "0.0.0.0:9094"
+type ClusterConfig struct {
```

Comment:
To avoid mistakes in the future (until the deprecated flags won't be fully removed), I would suggest you to prefix all deprecated struct field names (not the YAML) in `MultitenantAlertmanagerConfig ` with `Deprecated`:

```
	DeprecatedClusterBindAddr      string              `yaml:"cluster_bind_address"`
	DeprecatedClusterAdvertiseAddr string              `yaml:"cluster_advertise_address"`
	DeprecatedPeers                flagext.StringSlice `yaml:"peers"`
	DeprecatedPeerTimeout          time.Duration       `yaml:"peer_timeout"`
```

_I left the comment here cause I can't comment there._